spyder/plugins/ipythonconsole/utils/kernelspec.py

```diff
@@ -187,7 +197,7 @@ def env(self):
 
         # Environment variables that we need to pass to the kernel
         env_vars.update({
-            'SPY_EXTERNAL_INTERPRETER': not default_interpreter,
+            'SPY_EXTERNAL_INTERPRETER': True,
```

Comment:
Probably we should change this to `not default_interpreter or self.is_environment`